packages/components/src/button/style.scss

```diff
@@ -12,11 +11,16 @@
 	height: $button-size;
 	align-items: center;
 	box-sizing: border-box;
-	padding: 6px 12px;
 	overflow: hidden;
 	border-radius: $radius-block-ui;
 	color: $dark-gray-primary;
 
+	// Inside a placeholder, these properties need higher than usual specificity.
+	.components-placeholder & {
```

Comment:
We are adding these styles to the button component that is not aware that it may be used inside the placeholder component. What if we do the other way around and add the styles to the placeholder?

Reply:
That was my first instinct. But in doing so, we are duplicating styles from the button component into the placeholder component.

So either the button component is aware of the placeholder component, or the placeholder component is aware of the button component. In my mind, both are equally bad in that they go against the componentization goal, but at least by making the button aware, we don't have to duplicate styles.

The long term term solution, I believe, is to protect the `Placeholder` component using Shadow DOM. I believe this is exactly what Shadow DOM is created for — componentization for reuse in arbitrary environments. But absent that, we have to just be aware of the CSS bleed that is currently possible.

An alternate approach is to increase the specificity of the the font size and padding in the button component full stop — but I worry that will cause a cascading effect of small issues where every other element that currently uses the button component has to redo their CSS to increase specificity. Menu items, close buttons, you name it, especially the padding would likely cause havoc.

Reply:
> That was my first instinct. But in doing so, we are duplicating styles from the button component into the placeholder component.

I think that's fine because it communicates the intent more properly.

Reply:
I just realized that the Replace button in the toolbar suffers from the same:

<img width="706" alt="Screenshot 2020-05-12 at 14 53 03" src="https://user-images.githubusercontent.com/1204802/81693527-55793700-9460-11ea-9d80-c523b2825965.png">

Does that change the approach? And if not, would you mind if I add cases for both the placeholder and the toolbar buttons in the button component?

Reply:
This seems like something that can happen on multiple UI elements, not just placeholders and MediaReplace. I'm not sure what's the best approach to be honest. 

While I'd have been ok for the placeholder awareness to be in the button since they share the same package (componentts), it's not the case for MediaReplace. This is probably an issue because of the way we apply editor styles adds specificity causing these styles to have priority over the components styles.

I still feel fixing these per use-case is still the approach to go but It's not ideal :(

Reply:
I keep coming back to Shadow DOM as the best way to solve it. I know you think of Shadow DOM as "for bigger things", but in all the examples I read about Shadow DOM it's being used on a small-component basis, like encapsulating a slider but allowing a color variable to style it. Or encapsulating a button similarly.